pkg/provider/provider_test.go

```diff
@@ -209,7 +209,7 @@ lKn75l/9h0PwiiPaI0TGKN2O8AwvhGGwDElmFhYtXedbbaST6rbVRDUj
 			if err != nil {
 				t.Fatalf("expected nil err, got: %v", err)
 			}
-			if content != tc.expectedKey+tc.expectedCert {
+			if content != tc.expectedKey+certKeySeparator+tc.expectedCert {
```

Comment:
Could you also add a test which validates split on `certKeySeparator`

Reply:
added unit test